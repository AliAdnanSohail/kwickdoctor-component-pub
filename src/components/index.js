export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
