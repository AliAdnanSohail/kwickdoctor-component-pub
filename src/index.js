export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
