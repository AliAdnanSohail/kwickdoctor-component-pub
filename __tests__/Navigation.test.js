import React, { Component } from 'react';
import { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import { Navigation, NavigationItem } from '../src';

const Dashboard = () => <div>Dashboard component</div>;
const Notifications = () => <div>Notifications component</div>;

class NavigationWrapper extends Component {
  render() {
    return (
      <BrowserRouter>
        <main style={{ minHeight: '100vh', display: 'flex' }}>
          <Navigation>
            <NavigationItem route="/dashboard" icon="dashboard" label="Dashboards" />
            <NavigationItem route="/notifications" icon="notification_important" label="Notifications" />
          </Navigation>
          <Switch>
            <Route path="/dashboard" component={Dashboard} exact />
            <Route path="/notifications" component={Notifications} exact />
          </Switch>
        </main>
      </BrowserRouter>
    );
  }
}

it('renders correctly', () => {
  const element = mount(<NavigationWrapper />);

  expect(EnzymeToJson(element)).toMatchSnapshot();
});
