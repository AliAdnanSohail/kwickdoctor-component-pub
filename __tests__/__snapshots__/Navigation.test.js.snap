// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<NavigationWrapper>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <main
        style={
          Object {
            "display": "flex",
            "minHeight": "100vh",
          }
        }
      >
        <Navigation>
          <nav
            className="jsx-3540150986 nav"
          >
            <ul
              className="jsx-3540150986 nav__list"
            >
              <NavigationItem
                icon="dashboard"
                label="Dashboards"
                route="/dashboard"
              >
                <li>
                  <NavLink
                    activeClassName="nav-item--active"
                    ariaCurrent="true"
                    className="nav-item"
                    to="/dashboard"
                  >
                    <Route
                      path="/dashboard"
                    >
                      <Link
                        aria-current={false}
                        className="nav-item"
                        replace={false}
                        to="/dashboard"
                      >
                        <a
                          aria-current={false}
                          className="nav-item"
                          href="/dashboard"
                          onClick={[Function]}
                        >
                          <MaterialIcon
                            color="#ffffff"
                            icon="dashboard"
                            size={24}
                          >
                            <i
                              className="material-icons 24 md-dark "
                              style={
                                Object {
                                  "color": "#ffffff",
                                  "fontSize": 24,
                                }
                              }
                            >
                              dashboard
                            </i>
                          </MaterialIcon>
                          <span
                            className="nav-item__label"
                          >
                            Dashboards
                          </span>
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                  <style>
                    .nav-item{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:53px;height:53px;padding:16px;color:#ffffff;-webkit-text-decoration:none;text-decoration:none;-webkit-letter-spacing:0.15rem;-moz-letter-spacing:0.15rem;-ms-letter-spacing:0.15rem;letter-spacing:0.15rem;text-transform:uppercase;border-radius:50%;box-sizing:border-box;-webkit-transition:background-color 0.3s ease-in-out;transition:background-color 0.3s ease-in-out;font-size:0.75rem;font-weight:bold;}
                    .nav-item--active{background-color:#0c97f9;}
                    .nav-item .material-icons{width:21px;height:21px;}
                    .nav-item__label{display:none;}
                    @media (min-width:960px){.nav-item{width:100%;padding:16px 16px 16px 50px;border-top-left-radius:0;border-bottom-left-radius:0;border-top-right-radius:30px;border-bottom-right-radius:30px;}.nav-item .material-icons{margin-right:30px;}.nav-item__label{display:inline-block;}}
                  </style>
                </li>
              </NavigationItem>
              <NavigationItem
                icon="notification_important"
                label="Notifications"
                route="/notifications"
              >
                <li>
                  <NavLink
                    activeClassName="nav-item--active"
                    ariaCurrent="true"
                    className="nav-item"
                    to="/notifications"
                  >
                    <Route
                      path="/notifications"
                    >
                      <Link
                        aria-current={false}
                        className="nav-item"
                        replace={false}
                        to="/notifications"
                      >
                        <a
                          aria-current={false}
                          className="nav-item"
                          href="/notifications"
                          onClick={[Function]}
                        >
                          <MaterialIcon
                            color="#ffffff"
                            icon="notification_important"
                            size={24}
                          >
                            <i
                              className="material-icons 24 md-dark "
                              style={
                                Object {
                                  "color": "#ffffff",
                                  "fontSize": 24,
                                }
                              }
                            >
                              notification_important
                            </i>
                          </MaterialIcon>
                          <span
                            className="nav-item__label"
                          >
                            Notifications
                          </span>
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                  <style>
                    .nav-item{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:53px;height:53px;padding:16px;color:#ffffff;-webkit-text-decoration:none;text-decoration:none;-webkit-letter-spacing:0.15rem;-moz-letter-spacing:0.15rem;-ms-letter-spacing:0.15rem;letter-spacing:0.15rem;text-transform:uppercase;border-radius:50%;box-sizing:border-box;-webkit-transition:background-color 0.3s ease-in-out;transition:background-color 0.3s ease-in-out;font-size:0.75rem;font-weight:bold;}
                    .nav-item--active{background-color:#0c97f9;}
                    .nav-item .material-icons{width:21px;height:21px;}
                    .nav-item__label{display:none;}
                    @media (min-width:960px){.nav-item{width:100%;padding:16px 16px 16px 50px;border-top-left-radius:0;border-bottom-left-radius:0;border-top-right-radius:30px;border-bottom-right-radius:30px;}.nav-item .material-icons{margin-right:30px;}.nav-item__label{display:inline-block;}}
                  </style>
                </li>
              </NavigationItem>
            </ul>
            <JSXStyle
              css={
                Array [
                  ".nav.jsx-3540150986{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;height:100%;min-height:100vh;width:84px;padding:40px 0 80px 0;box-sizing:border-box;background-color:#354151;}",
                  ".nav__list.jsx-3540150986{-webkit-flex:1;-ms-flex:1;flex:1;list-style-type:none;padding:0 16px;margin:0;}",
                  "@media (min-width:960px){.nav.jsx-3540150986{width:330px;}.nav__list.jsx-3540150986{padding:0 50px 0 0;}}",
                ]
              }
              styleId="3540150986"
            />
          </nav>
        </Navigation>
        <Switch />
      </main>
    </Router>
  </BrowserRouter>
</NavigationWrapper>
`;
